### PR TITLE
basin delineation

### LINKED ISSUE #4: basin_mask update
- use basin/subbasin/interbasin syntax

### PR DESCRIPTION
**Pull request by DirkEilander**
* refactor basin delineation > bump pyflwdir version v0.4.7 > fixes #4 
* add basin delineation tests
* add example notebook

### CODE REVIEW

**Comment by DirkEilander** [comment]
Tests are failing because pyflwdir 0.4.7 is not yet found
It is already online though: https://anaconda.org/conda-forge/pyflwdir
Let's retry tomorrow

**Comment by DirkEilander** [comment]
- [ ] update the docs

File: envs/hydromt-dev.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: hydromt-dev
+name: hydromt-dev2
```
Comment (hboisgon) [inline envs/hydromt-dev.yml:1]:
```suggestion
name: hydromt-dev
```

File: hydromt/flw.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 logger = logging.getLogger(__name__)
 
 
-def flwdir_from_da(da, ftype="infer", check_ftype=True, mask=True):
+def flwdir_from_da(da, ftype="infer", check_ftype=True, mask=None):
```
Comment (hboisgon) [inline hydromt/flw.py:21]: This may have impacts on the plugins as well no? If yes might be good to add to the checklist of #1

Comment (hboisgon) [reply hydromt/flw.py:21]: Nevermind, I see you did a backwards compatibility as well

File: examples/delineate_basin.ipynb [hunk 3]
```diff
@@ -0,0 +1,399 @@
+{
```
Comment (hboisgon) [inline examples/delineate_basin.ipynb:1]: Really nice example notebook! A few typos so I'll commit an update.

File: envs/hydromt-dev.yml (hunk 1 above)
Comment (hboisgon) [reply envs/hydromt-dev.yml:1]: I will update the notebook in the end so will change it in the same commit

**Review by hboisgon — APPROVED** [review]
Nice developments! Both for the new basin_index and interbasin option.
I ran the notebook and it's really well done and helps go through these new developments.

I'll add a quick commit to update some typos in the notebook and change the dev environment name.

**Pull request merged**